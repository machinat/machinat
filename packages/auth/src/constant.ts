export const STATE_COOKIE_KEY = 'machinat_auth_state';
export const TOKEN_COOKIE_KEY = 'machinat_auth_token';
export const SIGNATURE_COOKIE_KEY = 'machinat_auth_signature';
export const ERROR_COOKIE_KEY = 'machinat_auth_error';

export const TOKEN_STORAGE_KEY = 'machinat_auth_token';
