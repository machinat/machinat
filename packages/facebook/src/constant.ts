export const FACEBOOK = 'facebook' as const;
export const FB = 'fb' as const;

export const PATH_PERSONAS = 'me/personas' as const;

export const PATH_FEED = 'me/feed' as const;
export const PATH_PHOTOS = 'me/photos' as const;
export const PATH_VIDEOS = 'me/videos' as const;
