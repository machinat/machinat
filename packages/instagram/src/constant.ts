export const INSTAGRAM = 'instagram' as const;
export const IG = 'ig' as const;
