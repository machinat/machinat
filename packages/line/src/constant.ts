export const CHANNEL_REQUEST_GETTER = Symbol(
  'channel_request_getter.line.sociably'
);
export const BULK_REQUEST_GETTER = Symbol('bulk_request_getter.line.sociably');

export const LINE = 'line' as const;

export const PATH_REPLY = 'v2/bot/message/reply';

export const PATH_PUSH = 'v2/bot/message/push';

export const PATH_MULTICAST = 'v2/bot/message/multicast';

export const PATH_RICHMENU = 'v2/bot/richmenu';

export enum LiffContextOs {
  Ios,
  Android,
  Web,
}
