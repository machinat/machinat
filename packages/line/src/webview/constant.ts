export const LIFF_ID_QUERY_KEY = 'liffId';
export const CHAT_CHANNEL_QUERY_KEY = 'chatChannelId';
export const GROUP_ID_QUERY_KEY = 'groupId';
export const ROOM_ID_QUERY_KEY = 'roomId';

export enum LiffOs {
  Ios,
  Android,
  Web,
}

export enum RefChatType {
  Utou,
  Group,
  Room,
  External,
  None,
}
