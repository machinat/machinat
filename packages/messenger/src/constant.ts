export const MESSENGER = 'messenger' as const;
export const FACEBOOK = 'facebook' as const;

export const ATTACHMENT_DATA = Symbol.for('attachment_data.messenger.sociably');
export const ATTACHMENT_INFO = Symbol.for('attachment_info.messenger.sociably');

export const ATTACHMENT_ASSET_TAG = Symbol.for(
  'attatchment_asset_tag.messenger.sociably'
);
export const API_PATH = Symbol.for('api_path.messenger.sociably');

export const MESSENGER_MESSAGING_TYPE_RESPONSE = 'RESPONSE';

export const PATH_MESSAGES = 'me/messages';

export const PATH_PASS_THREAD_CONTROL = 'me/pass_thread_control';

export const PATH_REQUEST_THREAD_CONTROL = 'me/request_thread_control';

export const PATH_TAKE_THREAD_CONTROL = 'me/take_thread_control';

export const PATH_MESSAGE_ATTACHMENTS = 'me/message_attachments';

export const PATH_MESSENGER_PROFILE = 'me/messenger_profile';

export const PATH_PERSONAS = 'me/personas';
