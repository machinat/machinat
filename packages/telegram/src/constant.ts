export const TELEGRAM = 'telegram' as const;
export const TG = 'tg' as const;
