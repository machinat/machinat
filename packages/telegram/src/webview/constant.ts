export const CHAT_ID_QUERY = 'chatId';
export const REDIRECT_QUERY = 'redirectUrl';
export const BOT_ID_QUERY = 'botId';

export const LOGIN_PARAMETERS = [
  'auth_date',
  'first_name',
  'id',
  'last_name',
  'photo_url',
  'username',
];
