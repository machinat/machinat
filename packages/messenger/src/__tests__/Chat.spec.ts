import { MessengerChatType } from '../constant';
import MessengerUser from '../User';
import MessengerChat from '../Chat';

test('from id', () => {
  const chat = new MessengerChat(1234567890, { id: '_PSID_' });

  expect(chat.platform).toBe('messenger');
  expect(chat.typeName()).toBe('MessengerChat');
  expect(chat.uid).toMatchInlineSnapshot(`"messenger.1234567890.psid._PSID_"`);

  expect(chat.identifier).toBe('_PSID_');
  expect(chat.threadType).toBe('USER_TO_PAGE');
  expect(chat.targetType).toBe('psid');
  expect(chat.target).toEqual({ id: '_PSID_' });

  expect(chat.toJSONValue()).toMatchSnapshot();
  expect(MessengerChat.fromJSONValue(chat.toJSONValue())).toStrictEqual(chat);
});

test('group chat', () => {
  const chat = new MessengerChat(
    1234567890,
    { id: '_PSID_' },
    MessengerChatType.Group
  );

  expect(chat.uid).toMatchInlineSnapshot(`"messenger.1234567890.psid._PSID_"`);

  expect(chat.identifier).toBe('_PSID_');
  expect(chat.threadType).toBe('GROUP');
  expect(chat.targetType).toBe('psid');
  expect(chat.target).toEqual(null);

  expect(chat.toJSONValue()).toMatchSnapshot();
  expect(MessengerChat.fromJSONValue(chat.toJSONValue())).toStrictEqual(chat);
});

test('user to user chat', () => {
  const chat = new MessengerChat(
    1234567890,
    { id: '_PSID_' },
    MessengerChatType.UserToUser
  );

  expect(chat.uid).toMatchInlineSnapshot(`"messenger.1234567890.psid._PSID_"`);

  expect(chat.identifier).toBe('_PSID_');
  expect(chat.threadType).toBe('USER_TO_USER');
  expect(chat.targetType).toBe('psid');
  expect(chat.target).toEqual(null);

  expect(chat.toJSONValue()).toMatchSnapshot();
  expect(MessengerChat.fromJSONValue(chat.toJSONValue())).toStrictEqual(chat);
});

test('from user_ref', () => {
  const chat = new MessengerChat(1234567890, { user_ref: '_USER_REF_' });

  expect(chat.platform).toBe('messenger');
  expect(chat.typeName()).toBe('MessengerChat');
  expect(chat.uid).toMatchInlineSnapshot(
    `"messenger.1234567890.user_ref._USER_REF_"`
  );

  expect(chat.identifier).toBe('_USER_REF_');
  expect(chat.threadType).toBe('USER_TO_PAGE');
  expect(chat.targetType).toBe('user_ref');
  expect(chat.target).toEqual({ user_ref: '_USER_REF_' });

  expect(chat.toJSONValue()).toMatchSnapshot();
  expect(MessengerChat.fromJSONValue(chat.toJSONValue())).toStrictEqual(chat);
});

test('from phone_number', () => {
  const chat = new MessengerChat(1234567890, {
    phone_number: '+88888888888',
    name: { first_name: 'jojo', last_name: 'doe' },
  });

  expect(chat.platform).toBe('messenger');
  expect(chat.typeName()).toBe('MessengerChat');
  expect(chat.uid).toMatchInlineSnapshot(
    `"messenger.1234567890.phone_number.nRn5C+EX4/vdk02aEWYs2zV5sHM="`
  );

  expect(chat.identifier).toBe('nRn5C+EX4/vdk02aEWYs2zV5sHM=');
  expect(chat.threadType).toBe('USER_TO_PAGE');
  expect(chat.targetType).toBe('phone_number');
  expect(chat.target).toEqual({
    phone_number: '+88888888888',
    name: { first_name: 'jojo', last_name: 'doe' },
  });

  expect(chat.toJSONValue()).toMatchSnapshot();
  expect(MessengerChat.fromJSONValue(chat.toJSONValue())).toStrictEqual(chat);
});

test('from post_id', () => {
  const chat = new MessengerChat(1234567890, { post_id: '_POST_ID_' });

  expect(chat.platform).toBe('messenger');
  expect(chat.typeName()).toBe('MessengerChat');
  expect(chat.uid).toMatchInlineSnapshot(
    `"messenger.1234567890.post_id._POST_ID_"`
  );

  expect(chat.identifier).toBe('_POST_ID_');
  expect(chat.threadType).toBe('USER_TO_PAGE');
  expect(chat.targetType).toBe('post_id');
  expect(chat.target).toEqual({ post_id: '_POST_ID_' });

  expect(chat.toJSONValue()).toMatchSnapshot();
  expect(MessengerChat.fromJSONValue(chat.toJSONValue())).toStrictEqual(chat);
});

test('from comment_id', () => {
  const chat = new MessengerChat(1234567890, {
    comment_id: '_COMMENT_ID_',
  });

  expect(chat.platform).toBe('messenger');
  expect(chat.typeName()).toBe('MessengerChat');
  expect(chat.uid).toMatchInlineSnapshot(
    `"messenger.1234567890.comment_id._COMMENT_ID_"`
  );

  expect(chat.identifier).toBe('_COMMENT_ID_');
  expect(chat.threadType).toBe('USER_TO_PAGE');
  expect(chat.targetType).toBe('comment_id');
  expect(chat.target).toEqual({ comment_id: '_COMMENT_ID_' });

  expect(chat.toJSONValue()).toMatchSnapshot();
  expect(MessengerChat.fromJSONValue(chat.toJSONValue())).toStrictEqual(chat);
});

test('MessengerChat.fromUser()', () => {
  const user = new MessengerUser(1234567890, '_USER_ID_');

  expect(MessengerChat.fromUser(user)).toEqual(
    new MessengerChat(1234567890, { id: '_USER_ID_' })
  );
});
