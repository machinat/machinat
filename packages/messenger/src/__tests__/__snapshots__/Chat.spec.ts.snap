// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`from comment_id 2`] = `
Object {
  "pageId": 1234567890,
  "target": Object {
    "comment_id": "_COMMENT_ID_",
  },
  "type": 0,
}
`;

exports[`from id 2`] = `
Object {
  "pageId": 1234567890,
  "target": Object {
    "id": "_PSID_",
  },
  "type": 0,
}
`;

exports[`from phone_number 2`] = `
Object {
  "pageId": 1234567890,
  "target": Object {
    "name": Object {
      "first_name": "jojo",
      "last_name": "doe",
    },
    "phone_number": "+88888888888",
  },
  "type": 0,
}
`;

exports[`from post_id 2`] = `
Object {
  "pageId": 1234567890,
  "target": Object {
    "post_id": "_POST_ID_",
  },
  "type": 0,
}
`;

exports[`from user_ref 2`] = `
Object {
  "pageId": 1234567890,
  "target": Object {
    "user_ref": "_USER_REF_",
  },
  "type": 0,
}
`;

exports[`group chat 2`] = `
Object {
  "pageId": 1234567890,
  "target": Object {
    "id": "_PSID_",
  },
  "type": 2,
}
`;

exports[`user to user chat 2`] = `
Object {
  "pageId": 1234567890,
  "target": Object {
    "id": "_PSID_",
  },
  "type": 1,
}
`;
